docs/plugins/main.py

```diff
@@ -189,7 +189,7 @@ def add_tabs(match: re.Match[str]) -> str:
         else:
             return '\n\n'.join(output)
 
-    return re.sub(r'^(``` *py.*?)\n(.+?)^```(\s+(?:^\d+\. .+?\n)*)', add_tabs, markdown, flags=re.M | re.S)
+    return re.sub(r'^(``` *py.*?)\n(.+?)^```(\s+(?:^\d+\. .+\n)*)', add_tabs, markdown, flags=re.M | re.S)
```

Comment:
The current regex did not work correctly with:

``````markdown
```python
some_code  # (1)!
```

1. Some annotation
   spanning multiple lines
``````

"Some annotation" was parsed correctly, but "spanning multiple lines" was interpreted as normal text.

The regex also does not match nested code blocks, e.g.:

``````markdown
- some list element:
  ```python
  ...
  ```
``````

But it's too much of a pain and I don't want to lose any more time on this. As I mentioned in https://github.com/pydantic/pydantic/issues/10083, the regex approach is really not ideal but the mkdocs/markdown Python implementation design is quite limiting here.